agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -0,0 +1,133 @@
+from abc import abstractmethod
+
+import crocoddyl
+import mim_solvers
+import numpy as np
+import numpy.typing as npt
+
+from agimus_controller.mpc_data import OCPResults, OCPDebugData
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.ocp_param_base import OCPParamsBaseCroco
+from agimus_controller.factory.robot_model import RobotModelFactory
+
+
+class OCPBaseCroco(OCPBase):
+    def __init__(
+        self,
+        robot_model: RobotModelFactory,
+        ocp_params: OCPParamsBaseCroco,
+    ) -> None:
+        """Defines common behavior for all OCP using croccodyl. This is an abstract class with some helpers to design OCPs in a more friendly way.
+
+        Args:
+            robot_model (RobotModelFactory): All models of the robot.
+            ocp_params (OCPParamsBaseCroco): Input data structure of the OCP.
+        """
+        # Setting the robot model
+        self._robot_model = robot_model
+        self._rmodel = self._robot_model._rmodel
+        self._cmodel = self._robot_model._complete_collision_model
+        self._armature = self._robot_model._params.armature
+
+        # Stat and actuation model
+        self._state = crocoddyl.StateMultibody(self._rmodel)
+        self._actuation = crocoddyl.ActuationModelFull(self._state)
+
+        # Setting the OCP parameters
+        self._ocp_params = ocp_params
+
+        self._ocp_results = None
+
+    @property
+    def horizon_size(self) -> int:
+        """Number of time steps in the horizon."""
+        return self._ocp_params.horizon_size
+
+    @property
+    def dt(self) -> float:
+        """Integration step of the OCP."""
+        return self._ocp_params.dt
+
+    @abstractmethod
+    def create_running_model_list(self) -> list[crocoddyl.ActionModelAbstract]:
+        """Create the list of running models."""
+        pass
+
+    @abstractmethod
+    def create_terminal_model(self) -> crocoddyl.ActionModelAbstract:
+        """Create the terminal model."""
+        pass
+
```

Comment:
I think we can add here an abstract function that looks like this
```suggestion

    @abstractmethod
    def update_crocoddyl_problem(self,x0: npt.NDArray[np.float64]) -> None:
        """Update the Crocoddyl problem's references, weights and x0."""
        pass
```

Reply:
Yes we could do like this! Let's have a quick discussion asap

Reply:
I agree, and this function should take the `WeightedTrajectoryPoint` instance as input.